packages/engine/src/postprocessing/components/PostProcessingComponent.tsx

```diff
@@ -0,0 +1,572 @@
+            entity={entity}
+            effectKey={effectKey}
+            schema={schema}
+            effects={effects}
+            camera={camera}
+            uiEffects={postprocessingComponent.effects}
+            scene={scene}
+            lut1DEffectTexture={lut1DEffectTexture}
+            lut3DEffectTexture={lut3DEffectTexture}
+            velocityDepthNormalPass={velocityDepthNormalPass}
+            useVelocityDepthNormalPass={useVelocityDepthNormalPass}
+            normalPass={normalPass}
+            depthDownsamplingPass={depthDownsamplingPass}
+            composer={renderer.value.effectComposer as EffectComposer}
+          ></EffectReactor>
+        )
+      })}
+    </>
+  )
+}
+
+const EffectReactor = (props: {
+  entity: Entity
+  effectKey: string
+  schema: any
+  effects: State<Record<string, Effect>>
+  camera: State<ArrayCamera>
+  uiEffects: State<EffectPropsSchema>
+  scene: State<Scene>
+  lut1DEffectTexture: Texture | null
+  lut3DEffectTexture: Texture | null
+  velocityDepthNormalPass: any
+  useVelocityDepthNormalPass: StateMethods<boolean, unknown>
+  normalPass: State<CustomNormalPass>
+  depthDownsamplingPass: State<DepthDownsamplingPass>
+  composer: EffectComposer
+}) => {
+  const {
+    entity,
+    effectKey,
+    schema,
+    effects,
+    camera,
+    uiEffects,
+    scene,
+    lut1DEffectTexture,
+    lut3DEffectTexture,
+    velocityDepthNormalPass,
+    useVelocityDepthNormalPass,
+    normalPass,
+    depthDownsamplingPass,
+    composer
+  } = props
+  const lookUp = {
+    BloomEffect: BloomEffectProcess,
+    BrightnessContrastEffect: BrightnessContrastEffectProcess,
+    ChromaticAberrationEffect: ChromaticAberrationEffectProcess,
+    ColorAverageEffect: ColorAverageEffectProcess,
+    ColorDepthEffect: ColorDepthEffectProcess,
+    DepthOfFieldEffect: DepthOfFieldEffectProcess,
+    DotScreenEffect: DotScreenEffectProcess,
+    FXAAEffect: FXAAEffectProcess,
+    GlitchEffect: GlitchEffectProcess,
+    GridEffect: GridEffectProcess,
+    HueSaturationEffect: HueSaturationEffectProcess,
+    LUT1DEffect: LUT1DEffectProcess,
+    LUT3DEffect: LUT3DEffectProcess,
+    LensDistortionEffect: LensDistortionEffectProcess,
+    LinearTosRGBEffect: LinearTosRGBEffectProcess,
+    MotionBlurEffect: MotionBlurEffectProcess,
+    NoiseEffect: NoiseEffectProcess,
+    PixelationEffect: PixelationEffectProcess,
+    SMAAEffect: SMAAEffectProcess,
+    SSAOEffect: SSAOEffectProcess,
+    SSGIEffect: SSGIEffectProcess,
+    SSREffect: SSREffectProcess,
+    ScanlineEffect: ScanlineEffectProcess,
+    ShockWaveEffect: ShockWaveEffectProcess,
+    TRAAEffect: TRAAEffectProcess,
+    TextureEffect: TextureEffectProcess,
+    TiltShiftEffect: TiltShiftEffectProcess,
+    ToneMappingEffect: ToneMappingEffectProcess,
+    VignetteEffect: VignetteEffectProcess
+  }
+
+  useEffect(() => {
+    if (!schema.isActive) {
+      effects[effectKey].set(none)
+    }
+  }, [schema.isActive])
+
+  useEffect(() => {
```

Comment:
For the next changes we can probably just have schema.isActive as the dep and both create and destroy in one useEffect. I don't wanna break functionality if this works for now 😅